### PR TITLE
Fix: Geoip's country flag issue & $().fancybox is not a function

### PR DESCRIPTION
**Pull request by Littlegolden**
Discussion thread: https://discuss.flarum.org/d/26103/28

Fix:
1. The problem of display country flag of geoip in thumbnails.
2. The problem of display `<head>` images in thumbnails (e.g. `<meta name="twitter:image" content="https://ftp.bmp.ovh/imgs/2021/03/1579582c22e42d64.png">`).

see problem 2 at https://discuss.flarum.org.cn/d/2475

### CODE REVIEW

**Comment by phenomlab** [comment]
@Littlegolden have you tested this PR ? I see that you removed the reference to `this` in one of the commits. In my testing, taking this route renders the extension inoperable.

**Comment by Littlegolden** [comment]
I tested it on the console before, and I wrote the wrong selector here：`.not($(".PostMeta-ip a img"))`, it should be `.not($(".PostMeta-ip img"))`.

But I just modified the `extend.php` of fancybox and tested it. Regardless of whether there has `this`, there is still a certain chance of this error:

image

**Comment by phenomlab** [comment]
Ok. Thanks. I will review this and test. The error you quoted above is a failure of the fancybox library to bind, which is normally caused by the absence of `this`

**Review by Littlegolden — COMMENTED** [review]
I don't know why after adding the flag filter, the img in `<head>` is not displayed too. Well, at least we fixed the flag issue.

File: extend.php [hunk 1]
```diff
@@ -20,23 +20,24 @@
...
+            $(this).wrap("<a class=\"fancybox\" href='" + currentImage.attr("data-src") + "'></a>");
+        }
+        else {
+            $(this).wrap("<a class=\"fancybox\" href='" + currentImage.attr("src") + "'></a>");
+        }
+        if ( $(this).parent().hasClass('fancybox') ) {
+            $().fancybox({
+                selector: '.fancybox'
+            });
+        }
```
Comment (Littlegolden) [inline extend.php:39]: ### Fix: $(...).fancybox is not a function
do `$().fancybox` only if this post have `.fancybox` image.

File: extend.php [hunk 2]
```diff
@@ -32,10 +32,15 @@
...
-            });
+        try {
+            $().ready(function(){
+                $().fancybox({
+                    selector: '.fancybox'
+                });
+            })
+        } catch (e) {
+            console.error(e.name);
+            console.error(e.message);
```
Comment (Littlegolden) [inline extend.php:43]: use try catch

**Review by Littlegolden — COMMENTED** [review]
Fix: $(...).fancybox is not a function

**Pull request merged**